core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java

```diff
@@ -0,0 +1,515 @@
+    List<EObject> allNodes = new ArrayList<>();
+    allNodes.addAll(((DSemanticDiagram) context).getOwnedDiagramElements());
+    for (EObject aObject : allNodes) {
+      if (aObject instanceof DNodeContainer) {
+        if (((DNodeContainer) aObject).getTarget() instanceof DAnnotation) {
+          if (((DAnnotation) ((DNodeContainer) aObject).getTarget()).getSource().equals(TITLE_BLOCK)) {
+            DDiagramElement aNode = ((DDiagramElement) aObject);
+            EObject nodeTarget = aNode.getTarget();
+            if (nodeTarget instanceof DAnnotation && aNode instanceof DDiagramElement) {
+              visibleElements.put((DAnnotation) nodeTarget, aNode);
+            }
+          }
+        }
+      }
+    }
+    for (Entry<DAnnotation, DDiagramElement> me : visibleElements.entrySet()) {
+      if (!selectedTitleBlocks.contains(me.getKey())) {
+        EObject container = me.getValue().eContainer();
+        if (container instanceof DSemanticDiagram) {
+          DAnnotation annotation = ((DAnnotation) (me.getValue().getTarget()));
+          if (isDiagram) {
+            if (isDiagramTitleBlock(annotation)) {
+              annotation.getDetails().put(VISIBILITY, FALSE);
+              DiagramServices.getDiagramServices().removeAbstractDNodeView((DNodeContainer) me.getValue());
+            }
+          } else {
+            if (!isDiagramTitleBlock(annotation)) {
+              annotation.getDetails().put(VISIBILITY, FALSE);
+              DiagramServices.getDiagramServices().removeAbstractDNodeView((DNodeContainer) me.getValue());
+            }
+          }
+        }
+      }
+    }
+    for (DAnnotation aTitleBlock : selectedTitleBlocks) {
+      if (!visibleElements.containsKey(aTitleBlock)) {
+
+        createTitleBlockView(context, aTitleBlock, diagram);
+      }
+    }
+    return context;
+  }
+
+  private AbstractDNode createTitleBlockView(EObject context, DAnnotation titleBlock, DDiagram diagram) {
+    String mappingName = IMappingNameConstants.CDB_TITLE_BLOCK;
+    NodeMapping mapping = DiagramServices.getDiagramServices().getNodeMapping(diagram, mappingName);
+    if (context instanceof DNodeList) {
+      if (null != titleBlock.getDetails().get(VISIBILITY)) {
+        titleBlock.getDetails().put(VISIBILITY, TRUE);
+      }
+      return DiagramServices.getDiagramServices().createDNodeListElement(mapping, titleBlock,
+          (DragAndDropTarget) context, diagram);
+    }
+    if (null != titleBlock.getDetails().get(VISIBILITY)) {
+      titleBlock.getDetails().put(VISIBILITY, TRUE);
+    }
+    return DiagramServices.getDiagramServices().createNode(mapping, titleBlock, (DragAndDropTarget) context, diagram);
+  }
+
+  public boolean isUniqueDiagramTitleBlock(EObject diagram) {
+    Collection<DAnnotation> result = new ArrayList<>();
+    if ((diagram instanceof DRepresentation)) {
+      DRepresentation representation = (DRepresentation) diagram;

```

Comment:
insetad of TITLE_BLOCK use DIAGRAM_TITLE_BLOCK.

Reply:
Done.